perses/app/setup_relative_calculation.py

```diff
@@ -1160,19 +1170,23 @@ def _generate_parsed_yaml(setup_options, input_yaml_file_path):
     """
     from openff.toolkit.topology import Molecule
     # The parsed yaml file will live in the experiment directory to avoid race conditions with other experiments
-    yaml_path = Path(setup_options['trajectory_directory'])
-    yaml_name = Path(input_yaml_file_path).name  # extract name from input/template yaml file.
+    yaml_path = AnyPath(setup_options['trajectory_directory'])
+    yaml_name = AnyPath(input_yaml_file_path).name  # extract name from input/template yaml file.
     time = datetime.datetime.now().strftime("%Y-%m-%d-%H:%M:%S")
     yaml_parse_name = f"perses-{time}-{yaml_name}"
     # Add timestamp information
     setup_options["timestamp"] = time
     # Read input sdf file and save into list -- We don't check stereochemistry
-    ligands_list = Molecule.from_file(setup_options['ligand_file'], allow_undefined_stereo=True)
+    ligand_file_path = AnyPath(setup_options['ligand_file'])
+    # Get the file format by getting the suffix and removing the "."
+    ligand_file_format = ligand_file_path.suffix[1:]
+    with open(ligand_file_path, "r") as ligand_file_object:
+        ligands_list = Molecule.from_file(ligand_file_object, file_format=ligand_file_format, allow_undefined_stereo=True)
     # Get names according to indices in parsed setup options
     setup_options['old_ligand_name'] = ligands_list[setup_options['old_ligand_index']].name
     setup_options['new_ligand_name'] = ligands_list[setup_options['new_ligand_index']].name
     # Write parsed and added setup options into yaml file
-    out_file_path = Path.joinpath(yaml_path, yaml_parse_name)
+    out_file_path = yaml_path / yaml_parse_name
```

Comment:
I don't think we want to join the paths this way, this leaves the possibility that if `yaml_path` is empty or `None` then it tries to write to the root filesystem, as in `/yaml_parse_name`. I think we want to use something like `Path.joinpath` or `os.path.join`.

Reply:
It is funny that you brought this up because I remember you doing the same thing at one point and it tripping me up :)
What is happening here is not some string manipulation, but with two path objects, you can use a `/` to make a path that respects the OS separator.

```
Python 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:25:59) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pathlib import Path
>>> foo = Path("some/path")
>>> bar = Path("other/place")
>>> foo / bar
PosixPath('some/path/other/place')
>>> 
```
RE: "`yaml_path` being empty" it is a required argument for our CLI so I don't think that is an issue.

Reply:
Ah I see, it's overloading the operator for the `Path` objects, okay. As long as we can guarantee they are like that this should be fine It is just that I recently had to fix a bug that I introduced in openmmtools, which was solved by using `os.path.join`, that's why I had it fresh in my memory. I think this should be okay!